runtime/src/append_vec.rs

```diff
@@ -594,6 +594,17 @@ impl AppendVec {
         ))
     }
 
+    /// Return account metadata for the account at `offset` if its data doesn't overrun
+    /// the internal buffer. Otherwise return None.
+    pub fn get_account_meta(&self, offset: usize) -> Option<AccountMeta> {
+        // Skip over StoredMeta data in the account
+        let offset = offset.checked_add(mem::size_of::<StoredMeta>())?;
+        // u64_align! does an unchecked add for alignment. Check that it won't cause an overflow.
```

Comment:
is this a danger in the current callers of `u64_align` right now like https://github.com/solana-labs/solana/blob/9ed4eac412616f2cb2e249a739bc4e2d4717f823/runtime/src/append_vec.rs#L516

Reply:
On a side note it might be good to extract all of these "align and offset" calculation functions into a common utility that can be reused in places like this: https://github.com/solana-labs/solana/blob/9ed4eac412616f2cb2e249a739bc4e2d4717f823/runtime/src/append_vec.rs#L511-L516 into a common utility so that we can reuse  them across these functions

Reply:
> is this a danger in the current callers of `u64_align` right now like
> 
> https://github.com/solana-labs/solana/blob/9ed4eac412616f2cb2e249a739bc4e2d4717f823/runtime/src/append_vec.rs#L516

Yes. In my test, if `next` is `usize::MAX`, it'll panic.

Reply:
> On a side note it might be good to extract all of these "align and offset" calculation functions into a common utility that can be reused in places like this:
> 
> https://github.com/solana-labs/solana/blob/9ed4eac412616f2cb2e249a739bc4e2d4717f823/runtime/src/append_vec.rs#L511-L516
> 
> into a common utility so that we can reuse them across these functions

Yes, that will be ideal. Separate PR though?

Reply:
I'd vote for separate pr. @alessandrod is good at getting this right. And @brooksprumo